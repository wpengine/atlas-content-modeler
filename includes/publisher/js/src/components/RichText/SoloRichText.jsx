import React from "react";
import RichTextHeader from "./RichTextHeader";

const SoloRichText = ({ modelSlug, field, fieldId }) => {
	return (
		<>
			<RichTextHeader modelSlug={modelSlug} field={field} />
			<textarea
				name={`atlas-content-modeler[${modelSlug}][${field.slug}]`}
				id={fieldId}
				defaultValue={field.value}
			/>
		</>
	);
};

export default SoloRichText;
