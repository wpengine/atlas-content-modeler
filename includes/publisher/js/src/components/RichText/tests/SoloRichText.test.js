import React from "react";
import renderer from "react-test-renderer";
import SoloRichText from "../SoloRichText";

describe("SoloRichText", () => {
	const props = {
		modelSlug: "cats",
		field: {
			slug: "name",
			name: "Name",
			description: "The cat's real name.",
			value: "Kahn The Destroyer",
		},
		fieldId: "field-123",
	};
	const TestComponent = renderer.create(<SoloRichText {...props} />);

	it("renders a matching snapshot", () => {
		expect(TestComponent.toJSON()).toMatchSnapshot();
	});
});
