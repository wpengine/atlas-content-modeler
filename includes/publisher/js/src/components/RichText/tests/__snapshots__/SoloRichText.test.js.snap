// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoloRichText renders a matching snapshot 1`] = `
Array [
  <label
    htmlFor="atlas-content-modeler[cats][name]"
  >
    Name
  </label>,
  <p
    className="help mb-4"
  >
    The cat's real name.
  </p>,
  <textarea
    defaultValue="Kahn The Destroyer"
    id="field-123"
    name="atlas-content-modeler[cats][name]"
  />,
]
`;
